### PR TITLE
[BugFix] Support count(1)/count(*)/count(col) rewrite when col is not nullable

### LINKED ISSUE #27715: Materialized rewrite generates bad plan for count(*)
### Steps to reproduce the behavior (Required)

On SSB dataset:

```sql
CREATE MATERIALIZED VIEW `mv_shipmode`
DISTRIBUTED BY RANDOM
REFRESH ASYNC
AS SELECT `lineorder`.`lo_shipmode`, count(`lineorder`.`lo_shipmode`) AS `count(lo_shipmode)`
FROM `ssb_1g`.`lineorder`
GROUP BY `lineorder`.`lo_shipmode`;
```

query: 
```sql
-- success
select count(lo_shipmode) from lineorder where lo_shipmode = 'MAIL';

-- failed: Vectorized engine does not support the operator, node_type
select count(*) from lineorder where lo_shipmode = 'MAIL'
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by LiShuMing**
Fixes #27715

fix some bugs:
- Sync MV should not support expr(aggregate func) syntax;
- Async MV should take whether scalarOp is rewritten in `rewriteScalarOperator `.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 4 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/CreateMaterializedViewStmt.java|2|3|66.67%|[423]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 4 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/CreateMaterializedViewStmt.java|2|3|66.67%|[423]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 21 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/EquationRewriter.java|17|17|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateMaterializedViewStmt.java|3|3|100.00%|[]|

**Review by murphyatwork — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 22 / 22 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/EquationRewriter.java|18|18|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateMaterializedViewStmt.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 24 / 24 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/EquationRewriter.java|20|20|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateMaterializedViewStmt.java|3|3|100.00%|[]|

**Review by murphyatwork — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/EquationRewriter.java [hunk 1]
```diff
@@ -178,23 +188,47 @@ private boolean replaceColInExpr(ScalarOperator expr, ColumnRefOperator oldCol,
...
     }
 
+    private static CallOperator normalizeCallOperator(CallOperator aggFunc) {
+        String aggFuncName = aggFunc.getFnName();
+        if (!aggFuncName.equals(FunctionSet.COUNT) || aggFunc.isDistinct()) {
+            return aggFunc;
+        }
+
+        // unify count(*) or count(non-nullable column) to count(1) to be better for rewrite.
+        if (aggFunc.getChildren().size() == 0 || !aggFunc.getChild(0).isNullable()) {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/EquationRewriter.java:198]: what is the aggFunc is already count(1)?

**Review by ABingHuang — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**